service/history/workflow/update/update.go

```diff
@@ -90,6 +95,17 @@ type (
 	}
 
 	updateOpt func(*Update)
+
+	UpdateStatus struct {
+		Stage   enumspb.UpdateWorkflowExecutionLifecycleStage
+		Outcome *updatepb.Outcome
+	}
+)
+
+const (
+	unspecifiedStage = enumspb.UPDATE_WORKFLOW_EXECUTION_LIFECYCLE_STAGE_UNSPECIFIED
+	acceptedStage    = enumspb.UPDATE_WORKFLOW_EXECUTION_LIFECYCLE_STAGE_ACCEPTED
+	completedStage   = enumspb.UPDATE_WORKFLOW_EXECUTION_LIFECYCLE_STAGE_COMPLETED
 )
```

Comment:
Looks like this is the second place these constants are defined and there are a bunch of places where you don't use these definitions.
Could you please use the enums directly to avoid the extra lookup?